pkg/operator/ceph/cluster/controller.go

```diff
@@ -388,8 +388,28 @@ func (c *ClusterController) requestClusterDelete(cluster *cephv1.CephCluster) (r
 		return reconcile.Result{}, nil // do not requeue the delete
 	}
 
-	logger.Infof("cleaning up CephCluster %q", nsName)
+	logger.Info("removing networkFence if matching cephCluster UID found")
+	networkFenceList := &addonsv1alpha1.NetworkFenceList{}
+	err := c.client.List(c.OpManagerCtx, networkFenceList)
```

Comment:
When calling List(), use a filter on the label in the query so that you don't get results for network fences from other clusters.

Reply:
done